modules/order/src/Entity/OrderItem.php

```diff
@@ -245,9 +267,16 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDescription(t('The number of purchased units.'))
       ->setReadOnly(TRUE)
       ->setSetting('unsigned', TRUE)
+      ->setSetting('precision', 12)
+      ->setSetting('scale', 3)
+      ->setSetting('max', '367384.999')
```

Comment:
We can put #min and #step settings here but we will be restricted by those settings on ModeNumberWidget form. For example I've added #max =  '367384.999' here on a base field. This setting could be decreased on a form display mode settings form but not increased. That is not the matter for the #max because it is most possible value for precision 12, scale 3 field. Conversely, it's a good thing preventing some unnecessary errors on the form, but if put here #min = 1, for example, we could never set #min < 1 on a form display mode settings form, i.e.- values like 0.001 will be forbidden. The same issue with #step.